daemon/api_test.go

```diff
@@ -1576,21 +1610,27 @@ func (s *apiSuite) TestDisconnectPlugFailureNoSuchPlug(c *check.C) {
 	c.Check(err, check.IsNil)
 	c.Check(body, check.DeepEquals, map[string]interface{}{
 		"result": map[string]interface{}{
-			"message": `cannot disconnect plug "plug" from snap "producer", no such plug`,
+			"message": `cannot perform the following tasks:
+- Disconnect producer:plug from consumer:slot (cannot disconnect plug "plug" from snap "producer", no such plug)`,
```

Comment:
Ack, I think we could do that by looking up plug and slot, doing the sanity checks and then queuing the security refresh only (so connect/disconnect would just become some synchronous action followed by enqueued setup-snap-security. Is that what you were thinking about?